### DISCUSSION TITLE
no such file or directory, open next-server.js.nft.json

### DISCUSSION THREAD

**Discussion by tachang** [vercel/next.js#43849] [Help]
When running turbopack and building I get the following:

```
build: > Build error occurred
build: [Error: ENOENT: no such file or directory, open '/home/user/workspace/app/.next/next-server.js.nft.json'] {
build:   errno: -2,
build:   code: 'ENOENT',
build:   syscall: 'open',
build:   path: '/home/user/workspace/app/.next/next-server.js.nft.json'
build: }
build: ERROR: command finished with error: command (/home/user/workspace/app) npm run build exited (1)
```

The file is available in .next/cache/next-server.js.nft.json

How do I tell turbo this file's location?

**Comment by drewbitt** [comment]
I also get this using `turbotrace` and pnpm. I did not get this error once deleting my node_modules & switching to `npm install && npm run build`.

**Comment by Mortezakiaa** [comment]
im also have this error on build with turbopack on standalone build output
next build --turbo

   ▲ Next.js 16.0.7 (Turbopack)
   - Environments: .env
   - Experiments (use with caution):
     · optimizePackageImports

   Creating an optimized production build ...
 ✓ Compiled successfully in 45s
 ✓ Finished TypeScript in 86s
 ✓ Collecting page data using 3 workers in 1246.9ms
 ✓ Generating static pages using 3 workers (19/19) in 1318.7ms

> Build error occurred
Error: ENOENT: no such file or directory, open 'C:\Users\m.kia\Desktop\EPC\.next\server\middleware.js.nft.json'
    at ignore-listed frames {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'open',
  path: 'C:\\Users\\m.kia\\Desktop\\EPC\\.next\\server\\middleware.js.nft.json'
}

whats problem???
any help

**Comment by maelfosso** [comment]
@Mortezakiaa I also have the same problem when running `npm run build`
```
$ npm run build                                                         ✔  1m 3s  14:22:20 

> f4life-website@0.1.0 build
> next build

   ▲ Next.js 16.0.6 (Turbopack)

   Creating an optimized production build ...
 ✓ Compiled successfully in 3.3s
 ✓ Finished TypeScript in 1736.4ms    
 ✓ Collecting page data using 15 workers in 190.2ms    
 ✓ Generating static pages using 15 workers (4/4) in 161.0ms

> Build error occurred
Error: ENOENT: no such file or directory, open '/Users/maelfosso/Documents/Clients/F4Life/website/.next/server/middleware.js.nft.json'
    at ignore-listed frames {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/Users/maelfosso/Documents/Clients/F4Life/website/.next/server/middleware.js.nft.json'
}
```

**Reply by Mortezakiaa** [reply]
I found the problem
It's about your middleware or if you use next js 16 maybe this problem with your proxy.ts or middleware.ts
You must check this files

**Comment by IbrahimSwaiss** [comment]
The problem was using `middleware.ts` in NextJs 16. This file should be renamed to `proxy.ts`.
https://nextjs.org/docs/messages/middleware-to-proxy
